src/logger.ts

```diff
@@ -3,20 +3,35 @@ import fs from "fs";
 import { inspect } from "util";
 
 import { Disposable, window } from "vscode";
+import * as vscode from "vscode";
 
 import { disposeAll } from "./utils";
 
 export enum LogLevel {
+    /** Disables all logging. */
     none = 0,
     error = 1,
     warn = 2,
-    debug = 3,
+    info = 3,
+    debug = 4,
 }
 
 export interface ILogger {
     debug(...args: any[]): void;
+    info(...args: any[]): void;
     warn(...args: any[]): void;
     error(...args: any[]): void;
+    /**
```

Comment:
How does this work with the old `debug`/`warn` methods? Should we stick to those (and submit the uri to them), or remove them in favour of `log`? Which one should you use?

Reply:
Adding `uri` to the existing debug/warn/error methods would be very high-churn for a somewhat limited use-case. (Possibly could be done with typescript "overload" signature, but also not sure that's worth the effort).

Reply:
Is the docstring here not sufficient?

Reply:
If I were a new contributor adding a new feature, I would be unsure whether to use `logger.warn` or `logger.log(uri, LogLevel.warn)`

Reply:
I understand. But in this case I can't think of a general guidance. New contributors sometimes need to figure things out. If they choose the "wrong" one, the cost is negligible, in this cas.